js/checkboxgroup/Index.svelte

```diff
@@ -14,46 +15,30 @@
 	export let elem_classes: string[] = [];
 	export let visible = true;
 	export let value: (string | number)[] = [];
-	export let value_is_output = false;
-	export let choices: [string, number][];
+	export let choices: [string, string | number][];
 	export let container = true;
 	export let scale: number | null = null;
 	export let min_width: number | undefined = undefined;
 	export let label = gradio.i18n("checkbox.checkbox_group");
 	export let info: string | undefined = undefined;
-	export let show_label: boolean;
+	export let show_label = true;
 
 	export let loading_status: LoadingStatus;
-	export let interactive: boolean;
-
-	let old_value: (string | number)[] = value.slice();
+	export let interactive = true;
 
-	function toggleChoice(choice: string | number): void {
-		// update the old
-		if (old_value.includes(choice)) {
-			old_value.splice(old_value.indexOf(choice), 1);
+	function toggle_choice(choice: string | number): void {
+		if (value.includes(choice)) {
+			value = value.filter((v) => v !== choice);
 		} else {
-			old_value.push(choice);
+			value = [...value, choice];
 		}
-		old_value = [...old_value];
-	}
 
-	function handle_change(): void {
-		gradio.dispatch("change");
-		if (!value_is_output) {
-			gradio.dispatch("input");
-		}
+		gradio.dispatch("input");
```

Comment:
Its pretty easy to just dispatch `input` explicitly here, we don't need the confusing `value_is_output` logic.

Reply:
Woudln't this also dispatch `input` if the value changes programmatically instead of via user input?

Reply:
Well I tested it and it doesn't seem to, though I'm not totally sure why

Reply:
`toggle_choice` is only called when the `input` elements `change` event is fired. This doesn't fire when the value is set programmatically, only when a user changes it in the UI.